vendor/README.md

```diff
@@ -0,0 +1,319 @@
+      console.error('Travis error: ' + err.stack);
+      if (res) { console.error(''+res); }
+      badgeData.text[1] = 'invalid';
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+      var state = res.headers['content-disposition']
+                     .match(/filename="(.+)\.svg"/)[1];
+      badgeData.text[1] = state;
+      if (state === 'passing') {
+        badgeData.colorscheme = 'brightgreen';
+      } else if (state === 'failing') {
+        badgeData.colorscheme = 'red';
+      } else {
+        badgeData.text[1] = state;
+      }
+      sendBadge(format, badgeData);
+
+    } catch(e) {
+      badgeData.text[1] = 'invalid';
+      sendBadge(format, badgeData);
+    }
+  });
+}));
+```
+
+Before getting started, install the project dependencies if you haven't
+already:
+
+```
+npm i
+```
+
+We'll start by creating a new module, `vendor/travis.js`, using this
+boilerplate:
+
+```js
+'use strict';
+
+const Joi = require('joi');
+const ServiceTester = require('./runner/service-tester');
+
+const t = new ServiceTester('Travis', '/travis');
+module.exports = t;
+```
+
+We'll add all our tests to this ServiceTester object, which gets exported from
+the module. The two arguments to the constructor are the name of the service
+and its URI prefix. The tester will prepend it to the URIs you provide later,
+which saves copying and pasting.
+
+Next we'll add a test for the typical case.
+
+[![](https://img.shields.io/badge/build-passing-brightgreen.svg)]()
+
+The JSON format for this badge is `{ name: 'build', value: 'passing' }`.
+
+Here's what our first test looks like:
+
+```js
+t.create('build status on default branch')
+  .get('/rust-lang/rust.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: Joi.equal('build'),
+    value: Joi.equal('failing', 'passing', 'unknown')
+  }));
+```
+
+The `create()` method gives the tester a new test. The chained-on calls come
+from the API testing framework [IcedFrisby][]. Here's a [longer example][] and
+the complete [API guide][IcedFrisby API].
+
+`expectJSONTypes()` is an IcedFrisby method which accepts a [Joi][] schema.
+Joi is a validation library that is build into IcedFrisby which you can use to
+match based on a set of allowed strings, regexes, or specific values. You can
+refer to their [API reference][Joi API].
+
+Notice we don't have to specify `/travis` again, or even `localhost`. The test
+runner handles that for us.
+
+When defining an IcedFrisby test, typically you would invoke the `toss()`
+method, to register the test. However, this is not necessary when using
+ServiceTester.
+
+[IcedFrisby]: https://github.com/MarkHerhold/IcedFrisby
+[longer example]: https://github.com/MarkHerhold/IcedFrisby/#show-me-some-code
+[IcedFrisby API]: https://github.com/MarkHerhold/IcedFrisby/blob/master/API.md
+[Joi]: https://github.com/hapijs/joi
+[Joi API]: https://github.com/hapijs/joi/blob/master/API.md
+
+Run the test:
+
+```
+npm run test:vendor -- --only=travis
```

Comment:
This is a small "travis" above, we used thebig travis. Is this referring to `/travis` or the name?